Userland/pro.cpp

```diff
@@ -24,23 +24,107 @@
  */
 
+#include <AK/GenericLexer.h>
+#include <AK/LexicalPath.h>
 #include <AK/NumberFormat.h>
 #include <AK/SharedBuffer.h>
 #include <AK/URL.h>
 #include <LibCore/ArgsParser.h>
 #include <LibCore/EventLoop.h>
+#include <LibCore/File.h>
 #include <LibProtocol/Client.h>
 #include <LibProtocol/Download.h>
 #include <stdio.h>
 
+// FIXME: Move this somewhere else when it's needed (e.g. in the Browser)
+class ContentDispositionParser {
+public:
+    ContentDispositionParser(const StringView& value)
+    {
+        GenericLexer lexer(value);
+
+        lexer.consume_while([](auto c) { return is_whitespace(c); });
+
+        if (lexer.consume_specific("inline")) {
+            m_kind = Kind::Inline;
+            if (!lexer.is_eof())
+                m_might_be_wrong = true;
+            return;
+        }
+
+        if (lexer.consume_specific("attachment")) {
+            m_kind = Kind::Attachment;
+            if (lexer.consume_specific(";")) {
+                lexer.consume_while([](auto c) { return is_whitespace(c); });
+                if (lexer.consume_specific("filename=")) {
+                    m_filename = lexer.consume_quoted_string();
+                } else {
+                    m_might_be_wrong = true;
+                }
+            }
+            return;
+        }
+
+        if (lexer.consume_specific("form-data")) {
+            m_kind = Kind::FormData;
+            while (lexer.consume_specific(";")) {
+                lexer.consume_while([](auto c) { return is_whitespace(c); });
+                if (lexer.consume_specific("name=")) {
+                    m_name = lexer.consume_quoted_string();
+                } else if (lexer.consume_specific("filename=")) {
+                    m_filename = lexer.consume_quoted_string();
+                } else {
+                    m_might_be_wrong = true;
+                }
+            }
+
+            return;
+        }
+
+        // FIXME: Support 'filename*'
+        m_might_be_wrong = true;
+    }
+
+    enum class Kind {
+        Inline,
+        Attachment,
+        FormData,
+    };
+
+    const StringView& filename() const { return m_filename; }
+    const StringView& name() const { return m_name; }
+    Kind kind() const { return m_kind; }
+    bool might_be_wrong() const { return m_might_be_wrong; }
+
+private:
+    StringView m_filename;
+    StringView m_name;
+    Kind m_kind { Kind::Inline };
+    bool m_might_be_wrong { false };
+};
+
 int main(int argc, char** argv)
 {
     const char* url_str = nullptr;
+    const char* output_file = nullptr;
+    bool save_at_provided_name = false;
+    bool silent = false;
 
     Core::ArgsParser args_parser;
+    args_parser.add_option(output_file, "Output file path", "output", 'o', "file");
+    args_parser.add_option(save_at_provided_name, "Write to a file named as the remote file", nullptr, 'O');
+    args_parser.add_option(silent, "Hide all output", "silent", 's');
```

Comment:
This is typically named `-q`/`--quiet`. And I'm not convinced having it is a good idea either, compared to `2>/dev/null`. And it doesn't seem to be used anywhere?

Reply:
Oh I added it, decided against it, then I guess forgot to remove the option!